internal/cli/cmd/plugin/build/build.go

```diff
@@ -148,242 +131,6 @@ func (o *option) Execute() error {
 	return output.HandleString(o.writer, fmt.Sprintf("%s built successfully", manifest.Name))
 }
 
-func (o *option) executeNPMSteps() error {
-	if err := o.executeNPMInstall(); err != nil {
-		return err
-	}
-	return o.executeNPMBuild()
-}
-
-func (o *option) executeNPMInstall() error {
-	if o.skipNPMInstall {
-		return nil
-	}
-	if _, err := os.Stat(filepath.Join(o.cfg.FrontendPath, "node_modules")); os.IsNotExist(err) {
-		// Then run `npm ci` to install the dependencies
-		cmd := exec.Command("npm", "ci")
-		cmd.Dir = o.cfg.FrontendPath
-		// to get a more comprehensive error message, we need to capture the stdout & stderr.
-		var stdoutBuffer bytes.Buffer
-		cmd.Stdout = &stdoutBuffer
-		var stderrBuffer bytes.Buffer
-		cmd.Stderr = &stderrBuffer
-		cmd.Dir = o.cfg.FrontendPath
-		if execErr := cmd.Run(); execErr != nil {
-			return fmt.Errorf("unable to install the frontend dependencies: %w, stdout: %s, stderr: %s", execErr, stdoutBuffer.String(), stderrBuffer.String())
-		}
-	}
-	return nil
-}
-
-func (o *option) executeNPMBuild() error {
-	if o.skipNPMBuild {
-		return nil
-	}
-	cmd := exec.Command("npm", "run", "build")
-	// to get a more comprehensive error message, we need to capture the stdout & stderr.
-	var stdoutBuffer bytes.Buffer
-	cmd.Stdout = &stdoutBuffer
-	var stderrBuffer bytes.Buffer
-	cmd.Stderr = &stderrBuffer
-	cmd.Dir = o.cfg.FrontendPath
-	if err := cmd.Run(); err != nil {
-		return fmt.Errorf("unable to build the frontend: %w, stdout: %s, stderr: %s", err, stdoutBuffer.String(), stderrBuffer.String())
-	}
-	return nil
-}
-
-// prepareCueFiles "prepares" the CUE files for the plugin packaging.
-// More precisely, it does the necessary to rely on the "old" modules resolution, so that the plugin archive
-// is fully "self-contained" and schema evaluation doesn't need to fetch dependencies from an OCI registry.
-func (o *option) prepareCueFiles() (func(), error) {
-	// Check if we need to vendor the dependencies
-	// - Parse the module file into a CUE instance
-	ctx := cuecontext.New()
-	insts := load.Instances([]string{o.moduleFilePath}, nil)
-	module := ctx.BuildInstance(insts[0])
-	if err := module.Err(); err != nil {
-		return nil, fmt.Errorf("failed to load plugin module file: %w", err)
-	}
-
-	// - if no `deps` are present, skip vendoring
-	deps := module.LookupPath(cue.ParsePath("deps"))
-	if deps.Err() != nil {
-		logrus.Debugf("`deps` not found, no CUE dependencies to vendor")
-		return nil, nil
-	}
-
-	// Backup the original state of cue.mod/pkg if it exists*, and restore the original state on exit.
-	// *: even if a given plugin relies on new dependency management, it may also rely on other dependencies
-	// managed the old way. This is technically possible & supported by CUE, thus we want to handle this case.
```

Comment:
So looking at the new code we lost this capability to support such "hybrid" setup with your change. Given the situation it's an acceptable drawback imho, just wanted your ack that this is done on purpose?

Reply:
yeah I did it on purpose because I saw some issue with the hybrid mode, and I felt for the moment removing this possibility will simplify the situation and the issues that can come